File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.js
function solution(n, a) {
    a.sort((x, y) => x - y);

    let left = 0;
    let maxKeep = 0;

    for (let right = 0; right < n; ++right) {
        while (a[right] - a[left] >= n) {
            left++;
        }

        let currentKeep = right - left + 1;
        if (currentKeep > maxKeep) {
            maxKeep = currentKeep;
        }
    }

    let minMoves = n - maxKeep;
    return minMoves;
}

module.exports = solution;

File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.rs
pub fn solution(n: i32, a: &mut Vec<i32>) -> i32 {
    a.sort();

    let mut left = 0;
    let mut max_keep = 0;
    let n_usize = n as usize;

    for right in 0..n_usize {
        while a[right] - a[left] >= n {
            left += 1;
        }

        let current_keep = (right - left + 1) as i32;
        if current_keep > max_keep {
            max_keep = current_keep;
        }
    }

    let min_moves = n - max_keep;
    min_moves
}